Sources/MongoSwift/SDAM.swift

```diff
@@ -3,12 +3,21 @@ import Foundation
 
 /// A struct representing a network address, consisting of a host and port.
 public struct ServerAddress: Equatable {
+    public enum HostType: String {
+        case ipv4
+        case ip_literal
```

Comment:
in Swift, the style for enum cases is generally camelCase. If you need to specify the string version separately for the sake of `Codable` conformance, then you can do so inline:
```swift
enum HostType: String {
    case ipLiteral = "ip_literal"
}
```